ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRule.kt

```diff
@@ -529,6 +533,23 @@ public class WrappingRule :
         }
     }
 
+    private fun insertNewLineBeforeSemi(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val previousCodeLeaf = node.prevCodeLeaf()?.lastChildLeafOrSelf() ?: return
+        val nextCodeLeaf = node.nextCodeLeaf()?.firstChildLeafOrSelf() ?: return
+        if (previousCodeLeaf.treeParent.elementType == ENUM_ENTRY && nextCodeLeaf.elementType == RBRACE) {
+            // Allow
+            // enum class INDEX2 { ONE, TWO, THREE; }
```

Comment:
I see no reason to allow this exception. Or is it added because wrapping is now run before the no-semi rule which removes this redundant semicolon? If so, then document that in a comment.

Reply:
Yes as when both rules are run final code will be 
```
enum class INDEX2 { ONE, TWO, THREE
}
```

If `no-semi` rule is not run than the final code will be like below
```
enum class INDEX2 { ONE, TWO, THREE;
}
```

In both the cases, final code was not looking good